server/services/permissions/localpermissions/localpermissions.go

```diff
@@ -27,7 +27,9 @@ func New(store permissions.Store, logger *mlog.Logger) *Service {
 }
 
 func (s *Service) HasPermissionToTeam(userID, teamID string, permission *mmModel.Permission) bool {
-	// Locally there is no team, so return true
```

Comment:
Reverting the fix for flaky cypress test as it caused consistent failures for backend tests.  IMO teamID should never be empty.